src/mainwindow.cpp

```diff
@@ -525,3 +544,46 @@ void MainWindow::NeedChanged()
     mItemExplorer->RefreshAfterMaxOneItemDifference();
     mItemExplorerUnassigned->RefreshAfterMaxOneItemDifference();
 }
+
+void MainWindow::ItemFinder()
+{
+    QDir dataDir(QCoreApplication::applicationDirPath() + "/Data/Items");
+
+    if (!dataDir.exists())
+    {
+        QMessageBox::critical(this, "Azharja", "Data folder not found.");
+        return;
+    }
+
+    QStringList filters;
+    filters  << "*.txt";
+    dataDir.setNameFilters(filters);
+
+    bool okPressed;

```

Comment:
You should always initialize variables when you declare them, to avoid having undeterministic behavior.

Reply:
Done.